libsql-client/src/lib.rs

```diff
@@ -232,19 +229,23 @@ impl Connection {
     /// * `stmts` - SQL statements
     pub async fn batch(
         &self,
-        stmts: impl IntoIterator<Item = impl Into<String>>,
-    ) -> Result<Vec<ResultSet>> {
+        stmts: impl IntoIterator<Item = impl Into<Statement>>,
+    ) -> Result<Vec<QueryResult>> {
         let mut headers = Headers::new();
         headers.append("Authorization", &self.auth).ok();
-        let stmts: Vec<String> = stmts
-            .into_iter()
-            .map(|s| format!("\"{}\"", s.into()))
-            .collect();
+        // FIXME: serialize and deserialize with existing routines from sqld
+        let mut body = "{\"statements\": [".to_string();
```

Comment:
Just an idea:
Would
```
body += itertools::join(stmts.into_iter().map(|s| format!("{}", s.into())), ",");
body += "]}";
```
work?

Reply:
It would, but it would also add `itertools` as a dependency, and cloudflare workers have a hard 1MiB limit on total binary size, because they store the WebAssembly-compiled code on their instances, so I'd rather leave it as is